fsspec/caching.py

```diff
@@ -256,12 +303,6 @@ def __init__(
         self.maxblocks = maxblocks
         self._fetch_block_cached = functools.lru_cache(maxblocks)(self._fetch_block)
 
-    def __repr__(self) -> str:
```

Comment:
There is an argument that repr should give details of what the object *is* (ideally, how to recreate it), but str should be the formatted version with details.

Reply:
This was moved to the parent class so all the cache implementations have a clearer picture of what they are and their current state, currently only some implementations have a `__repr__`.

Reply:
Yes, right. The comment came here to show how these reprs used to be, more in line with the typical python `<MyClass: details>` style.

Reply:
Ah OK, should we use this convention? something like this (note that `total_returned_bytes` is not tracked yet)

```python
        return f"""
            <{self.__class__.__name__}:
            block size  :   {self.blocksize}
            block count :   {self.nblocks}
            file size   :   {self.size}
            cache hits  :   {self.hit_count}
            cache misses:   {self.miss_count}
            total requested bytes: {self.total_requested_bytes}
            total returned bytes: {self.total_returned_bytes}>
        """
```

I was thinking of using `rich` to have the rich representation, a bar with colors to represent which parts of the file have been cached or something easy to visualize.

Reply:
Rich or HTML (for jupyter) would be nice, but fsspec shouldn't depend on either. There could be a separate pprint or similar method for that.

> the “official” string representation of an object. If at all possible, this should look like a valid Python expression that could be used to recreate an object with the same value (given an appropriate environment). If this is not possible, a string of the form <...some useful description...> should be returned.

https://docs.python.org/3/reference/datamodel.html#object.__repr__

Reply:
Yeah that would be for a future PR, do you think the proposed  `__repr__` above fits the bill of the convention `<MyClass: details>`?